### PR TITLE
Fixed issue #19087: Account takeover through Import User function

### PR DESCRIPTION
**Pull request by Shnoulle**
Dev: check permission for each line

### CODE REVIEW

File: application/controllers/UserManagementController.php [hunk 1]
```diff
@@ -933,6 +933,16 @@ public function actionImportUsers(string $importFormat = 'csv'): string
 
             if ($oUser  !== null) {
                 if ($overwriteUsers) {
+                    /* Check permission to edit this user */
+                    if (!$oUser->canEdit() || $aUser['uid'] == App()->user->id) {
+                        Yii::app()->setFlashMessage(sprintf(gT("You don't have permission to edit user %s."), $aNewUser['users_name']), 'warning');
+                        continue;
+                    }
+                    /* Check permission to edit self */
+                    if ($aUser['uid'] == App()->user->id) {
```
Comment (gabrieljenik) [inline application/controllers/UserManagementController.php:942]: Isn't this if statement overlapping witht he previous one?

File: application/models/SurveysGroupsettings.php [hunk 2]
```diff
@@ -418,7 +418,12 @@ protected static function translateOptionLabels($instance, $attribute, $value)
         } elseif ($attribute == 'owner_id' && $value != -1) {
             $instance->oOptions->owner = "";
             $instance->oOptions->ownerLabel = "";
-            $oUser = User::model()->findByPk($instance->oOptions->{$attribute});
+            /* \User|false[] see mantis #19426 */
```
Comment (gabrieljenik) [inline application/models/SurveysGroupsettings.php:421]: Is this expected here?

Comment (Shnoulle) [reply application/models/SurveysGroupsettings.php:421]: Lakh of @var

Comment (Shnoulle) [reply application/models/SurveysGroupsettings.php]: `/* @var array $oStaticUsers \User or null see mantis #19426 */`

File: application/models/SurveysGroupsettings.php [hunk 3]
```diff
@@ -418,7 +418,12 @@ protected static function translateOptionLabels($instance, $attribute, $value)
         } elseif ($attribute == 'owner_id' && $value != -1) {
             $instance->oOptions->owner = "";
             $instance->oOptions->ownerLabel = "";
-            $oUser = User::model()->findByPk($instance->oOptions->{$attribute});
+            /* @var array $oStaticUsers \User or null see mantis #19426 */
```
Comment (gabrieljenik) [inline application/models/SurveysGroupsettings.php:421]: These lines belong to 19087 or to 19426?

**Review by gabrieljenik — APPROVED** [review]
Code looks good.
Extra: A small user cache was implemented when rendering SurveySettings.

**Pull request merged**